lib/settings.js

```diff
@@ -120,7 +135,7 @@ module.exports = {
                 obj.ssl_key = true;
             }
             res.send(obj);
-            return next();
+            next(err);
```

Comment:
You'll want to check/call `next(err)` before here - as in the case of an error, it would be bad form to call both res.send() and next(err).

Reply:
Yeah, good catch. I've added a lot of error handling completely ignored across just the files I've needed to modify due to s/async/vasync/ dependency replacement. Which makes me wonder how many are there like these across the whole AdminUI code base ...

Reply:
Yeah, I saw a lot of cb fixes you added to the unsafe practices before - good work, it's a step in the right direction :)